newlib/libc/machine/riscv/memcpy-asm.S

```diff
@@ -28,5 +28,91 @@ memcpy:
 2:
   ret
 
-  .size	memcpy, .-memcpy
+#else
+.Lwordify:
+    andi   a3, a0, 3
+    beqz   a3, .Laligned
```

Comment:
if i see this correctly, you are always checking for 64-bit alignment, but then you only use 32-bit memory accesses